backend/pasta_curves/benches/bench_pasta_curves.rs

```diff
@@ -0,0 +1,114 @@
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::Measurement, BenchmarkGroup,
+    BenchmarkId, Criterion,
+};
+use group::{ff::Field, Curve, Group};
+use halo2_proofs::arithmetic::best_multiexp;
+use pasta_curves::{arithmetic::CurveAffine, pallas, vesta};
+
+fn bench_add_ff<G: Group, M: Measurement>(c: &mut BenchmarkGroup<'_, M>) {
+    let mut rng = rand::thread_rng();
+    c.bench_function("add_ff", |b| {
+        let lhs = G::Scalar::random(&mut rng);
+        let rhs = G::Scalar::random(&mut rng);
+        b.iter(|| black_box(lhs) + black_box(rhs))
+    });
+}
+
+fn bench_mul_ff<G: Group, M: Measurement>(c: &mut BenchmarkGroup<'_, M>) {
+    let mut rng = rand::thread_rng();
+    c.bench_function("mul_ff", |b| {
+        let lhs = G::Scalar::random(&mut rng);
+        let rhs = G::Scalar::random(&mut rng);
+        b.iter(|| black_box(lhs) * black_box(rhs))
+    });
+}
+
+fn bench_invert<G: Group, M: Measurement>(c: &mut BenchmarkGroup<'_, M>) {
+    let mut rng = rand::thread_rng();
+    c.bench_function("invert", |b| {
+        let a = G::Scalar::random(&mut rng);
+        b.iter(|| a.invert().unwrap());
+    });
+}
+
+fn bench_add_ec<G: Group, M: Measurement>(c: &mut BenchmarkGroup<'_, M>) {
+    let mut rng = rand::thread_rng();
+    c.bench_function("add_ec", |r| {
+        let a = G::random(&mut rng);
+        let b = G::random(&mut rng);
+        r.iter(|| a.add(b))
+    });
+}
+
+fn bench_dbl_ec<G: Group, M: Measurement>(c: &mut BenchmarkGroup<'_, M>) {
+    let mut rng = rand::thread_rng();
+    c.bench_function("dbl_ec", |r| {
+        let a = G::random(&mut rng);
+        r.iter(|| a.double())
+    });
+}
+
+fn bench_mul_ec<G: Group, M: Measurement>(c: &mut BenchmarkGroup<'_, M>) {
+    let mut rng = rand::thread_rng();
+    c.bench_function("mul_ec", |b| {
+        let lhs = G::random(&mut rng);
+        let rhs = G::Scalar::random(&mut rng);
+        b.iter(|| black_box(lhs) * black_box(rhs))
```

Comment:
In practice this benchmark can be reduced by 50-60% using `group::{WnafBase, WnafScalar}` (the precomputation cost of a window size of 6 pays off even just for a single multiplication), but I don't know if the `mul_ec` benchmark is meant to include performance improvements that require heap allocations:
```suggestion
        // The w-NAF implementation requires heap allocations:
        // - `WnafBase`: `2^(window_size - 1)` cached multiples of `G`.
        // - `WnafScalar`: At most `8 * scalar_byte_len` `i64` values.
        //
        // For Pallas and Vesta, `G` is 96 bytes and a scalar is 32 bytes, so the maximum
        // heap memory usage for a window size of 6 (ignoring overhead) is 5 kiB. This is
        // a nice trade-off between memory usage, precomputation cost, and performance
        // increase.
        let lhs = WnafBase::<_, 6>::new(G::random(&mut rng));
        let rhs = WnafScalar::new(&G::Scalar::random(&mut rng));
        b.iter(|| black_box(&lhs) * black_box(&rhs))
```